go/README.md

```diff
@@ -0,0 +1,117 @@
+
+A local version of caddy is in caddy/caddy. This is a vendored version of caddy which includes the plugins and middlewares for PacketFence.
+
+In order to build the caddy HTTP service:
+
+```
+# make pfcaddy
+# mv pfcaddy /usr/local/pf/bin/
+```
+
+## Creating a service
+
+Once you've built pfcaddy, you can use a Caddyfile to load your middleware and bind it on a specific port:
+
+```
+localhost:1234 {
+  logger {
+    requesthistory 100
+    level DEBUG
+  }
+  statsd {
+    proto udp
+    prefix pfsso
+  }
+  pfsso
+}
+```
+
+This file should be put in `/usr/local/pf/conf/caddy-services/NAME-OF-SERVICE.conf`
+
+Note how you can control the logger configuration from the Caddyfile. If your middleware (in this example pfsso) uses or calls the logger, you *must* declare it in your Caddyfile.
+
+If your middleware uses statsd, you don't have to configure statsd in your Caddyfile which will result in the packets just not being sent (a dummy statsd client will be created).
+
+You can start pfcaddy with your Caddyfile using the following command:
+
+```
+# /usr/local/pf/bin/pfcaddy -conf /usr/local/pf/conf/caddy-services/pfsso.conf
+```
+
+Once you have ascertained that the service is working correctly, you need to create an instance of pf::services::manager for it. You will also need to create a unitfile for it in conf/systemd like the following:
+
+```
+[Unit]
+Description=PacketFence Example Service
+Wants=packetfence-base.target packetfence-config.service packetfence-iptables.service
+After=packetfence-base.target packetfence-config.service packetfence-iptables.service
+Before=packetfence-pfdhcplistener.service
+
+[Service]
+PIDFile=/usr/local/pf/var/run/pfexample.pid
+ExecStart=/usr/local/pf/bin/pfcaddy -conf /usr/local/pf/conf/caddy-services/pfexample.conf
+Restart=on-failure
+Slice=packetfence.slice
+
+[Install]
+WantedBy=packetfence.target
+```
+
+Make sure that the packaging is also updated to copy those files in the /usr/lib/systemd/system directory.
+
+## Running the tests
+
+Like the perl unit tests, the Golang tests rely on the presence of the test pfconfig process to execute properly.
+
+In order to start the test pfconfig process:
+
+```
+# cd /usr/local/pf/t && ./pfconfig-test
+```
+
+Then you can proceed to execute all or some of the Golang unit tests:
+
+```
+# cd $GOPATH/src/github.com/inverse-inc/packetfence/go
+# go test ./...
+
+# cd $GOPATH/src/github.com/inverse-inc/packetfence/go/firewallsso/lib
+# go test
+```
+
+## TODO
+
+ * Integrate with pfconfig namespace expiration (for now any config change requires a restart)
+ * Integrate a vendoring solution
+ * I (Julien) don't like the name pfcaddy even though I came up with it. Suggestions on the name are welcome.
```

Comment:
I would use a dash in the name, just like we do for our other services.

How about 'pfappserver' ?